pom.xml

```diff
@@ -143,11 +143,39 @@
                 <spark.test.version>${spark321cdh.version}</spark.test.version>
                 <parquet.hadoop.version>1.10.1</parquet.hadoop.version>
                 <cloudera.repo.enabled>true</cloudera.repo.enabled>
+                <!-- Keeping the scala plugin version 4.3.0 for details
+                check https://github.com/NVIDIA/spark-rapids/issues/11112 -->
+                <scala.plugin.version>4.3.0</scala.plugin.version>
             </properties>
             <modules>
                 <module>shim-deps/cloudera</module>
                 <module>delta-lake/delta-20x</module>
             </modules>
+            <build>
+                <plugins>
+                    <plugin>
+                        <groupId>org.apache.maven.plugins</groupId>
+                        <artifactId>maven-enforcer-plugin</artifactId>
+                        <executions>
+                            <execution>
+                                <id>enforce-cloudera-jdk-version</id>
+                                <goals>
+                                    <goal>enforce</goal>
+                                </goals>
+                                <configuration>
+                                    <rules>
+                                        <requireJavaVersion>
+                                            <message>Only Java 8, 11 are supported for Cloudera! For details please
+                                                see https://github.com/NVIDIA/spark-rapids/issues/11112</message>
+                                            <version>(,17)</version>
+                                        </requireJavaVersion>
+                                    </rules>
+                                </configuration>
+                            </execution>
+                        </executions>
+                    </plugin>
+                </plugins>
+            </build>
```

Comment:
This applies to all Cloudera profiles. Thus instead of copy-and-pasting this should be moved to shim-deps/cloudera/pom.xml